### ISSUE TITLE
[Mentor] E-email de confirmação de desativação de conta

### ISSUE THREAD

**Issue by jonspada** [SouJunior/products#737]
**Nome:** User Story - Portal de Mentorias
**Assunto:**  [Mentor] E-mail de confirmação de desativação de conta
**Título:** Como mentor gostaria de receber um e-mail confirmando a desativação da minha conta do Portal de Mentorias
**Descrição:** Essa User Story mostra como será o e-mail de recebimento do mentor e também as tentativas em resgatar o mentor par plataforma, sendo ela 15 dias depois e outra 29 dias depois.

OBS - Task originária desta aqui : https://github.com/orgs/SouJunior/projects/11/views/1?filterQuery=desat&pane=issue&itemId=71295799&issue=SouJunior%7Cproducts%7C660

link das imagens em alta resolução: https://www.figma.com/design/HGa6jj4gz4MXyardmDjQ9z/Portal-de-Mentorias?node-id=8517-56912&node-type=canvas&t=Niqyv33HEtESic5w-0

(Imagem 1)

**Recebendo e-mail de confirmação de desativação**
O mentor irá receber um e-mail de confirmação de desativação de conta, e com a opção de estar fazendo a reativação,
ou também solicitar o suporte, clicando ali em azul ''suporte".

![Image](https://github.com/user-attachments/assets/7e775473-287e-46d5-8cbf-f01b6cfb6719)

___________________________________________________________________________________________________________

(Imagem 2)

**Segundo e-mail de resgate ao mentor 15 dias após o pedido de exclusão**

15 dias após o pedido de exclusão, é encaminhado ao mentor um e-mail de reengajamento para que ele retorno ao uso da plataforma. Havendo interesse em reativar sua conta, o mentor deve clicar no botão em azul "Realizar login e reativar conta'', ao fazer isso será redirecionado a tela de login do Portal de Mentorias.

![Image](https://github.com/user-attachments/assets/13f6bede-1ff7-4548-abe2-4855782b9645)

___________________________________________________________________________________________________________

(Imagem 3)

**Terceiro e-mail de resgate ao mentor 29 dias após o pedido de exclusão**
29 dias após o pedido de exclusão, é encaminhado ao mentor um e-mail de reengajamento para que ele retorno ao uso da plataforma. Havendo interesse em reativar sua conta, o mentor deve clicar no botão em azul "Realizar login e reativar conta'', ao fazer isso será redirecionado a tela de login do Portal de Mentorias.

![Image](https://github.com/user-attachments/assets/413e6659-4683-4a73-b968-c6e1e3e25d4f)

___________________________________________________________________________________________________________

(Imagem 4)

Quarto e-mail. Confirmação da desativação de conta.
É encaminhado um último e-mail confirmando que a conta do Mentor foi desativada, e, também, informando de que seu perfil não está mais disponível para futuros agendamentos. Caso mentor deseje continuar na plataforma, ele deve clicar em "Realizar login e reativar conta", ao fazer isso será redirecionado a tela de login do Portal de Mentorias.

![Image](https://github.com/user-attachments/assets/0a8dd5db-6262-492e-90f5-a8c07f21a1d1)

___________________________________________________________________________________________________________

**Critérios de Interface**

- [ ] O botão ou link "Realizar login e reativar conta" funciona corretamente, redirecionando o mentor para a tela de login do Portal de Mentorias em cada e-mail onde é disponibilizado.
- [ ] O texto "suporte" deve estar em formato de link sublinhado e redirecionar para a página de FAQ e Suporte do Portal de Mentorias ao ser clicado.
- [ ] O texto "suporte" ao ser destacado pelo usuário com o mouse (hover) deve mudar de cor ou ser destacado visualmente para indicar interatividade.
- [ ] Todos os e-mails são responsivos, adaptando-se automaticamente a diferentes tamanhos de tela (computador, tablet e celular).
- [ ] A formatação dos textos e links nos e-mails garante acessibilidade para leitores de tela e atende às práticas de usabilidade.

**Critérios de Aceite**

 **Recebimento do E-mail de Confirmação de Desativação**
- [ ] O mentor recebe um e-mail confirmando a desativação de sua conta no Portal de Mentorias imediatamente após solicitar a desativação.
- [ ] O e-mail contém um link funcional para reativação de conta, com texto "Realizar login e reativar conta".
- [ ] O link "suporte" sublinhado em azul redireciona o mentor para a página de FAQ e Suporte do Portal de Mentorias.

 **Primeiro E-mail de Reengajamento - 15 Dias Após o Pedido de Desativação**
- [ ] 15 dias após o pedido de exclusão, o mentor recebe um e-mail de reengajamento com o link "Realizar login e reativar conta", redirecionando para a tela de login do Portal de Mentorias.
- [ ] O e-mail explica que o mentor pode retomar o uso da plataforma caso clique no link.
- [ ] O design do e-mail é compatível com diferentes dispositivos (computador, tablet, celular).

**Segundo E-mail de Reengajamento - 29 Dias Após o Pedido de Desativação**
- [ ] 29 dias após o pedido de exclusão, é enviado um novo e-mail de reengajamento ao mentor, com o link "Realizar login e reativar conta" que redireciona para a tela de login.
- [ ] O e-mail reforça a possibilidade de reativação da conta e seu uso na plataforma.
- [ ] O design do e-mail é compatível com diferentes dispositivos (computador, tablet, celular).

 **Confirmação Final de Desativação - E-mail Definitivo**
- [ ] 30 dias após a solicitação de exclusão, é enviado um e-mail final de confirmação de desativação, informando que o perfil do mentor não estará mais disponível para futuros agendamentos.
- [ ] Este e-mail inclui uma opção de "Realizar login e reativar conta", redirecionando para a tela de login do Portal de Mentorias caso o mentor decida retornar.
- [ ] O design do e-mail é compatível com diferentes dispositivos (computador, tablet, celular).
___________________________________________________________________________________________________________

**DoR - Definition of Ready**
- [ ] O item ter sido refinado seguindo o modelo de User Story.

- [ ] A User Story deverá ter pelo menos um critério de aceite.

- [ ]  A User Story deverá ter pelo menos uma referência visual (Ex: Protótipo, Fluxograma e etc...).

- [ ] A User Story deverá apresentar o seu módulo de referência: Mentor ou Mentorado.

- [ ] Ter sido validado pelo mentor.
___________________________________________________________________________________________________________

**DoD - Definition of Done**
- [ ] Documentar ferramentas, documentação e ambientes necessários para realização de teste.

- [ ] Execução de testes unitários.

- [ ] Revisão do código do Pull request validado pelo mentor e por outro membro do time de desenvolvimento.

- [ ] Critérios de aceite verificados, testados e validado.

- [ ] Teste de software criados, executados e validados pelo QA.

- [ ] Garantir que todas as etapas planejadas foram atendidas/concluídas.
___________________________________________________________________________________________________________

link das imagens em alta resolução: https://www.figma.com/design/HGa6jj4gz4MXyardmDjQ9z/Portal-de-Mentorias?node-id=8517-56912&node-type=canvas&t=Niqyv33HEtESic5w-0

**Comment by DanielDefault1** [comment]
@jonspada você chegou a fazer os ajustes que a Amanda comentou no nosso último checkpoint?

**Comment by jonspada** [comment]
> [@jonspada](https://github.com/jonspada) você chegou a fazer os ajustes que a Amanda comentou no nosso último checkpoint?

Daniel acredito que fiz todas ali sim, se tiver alguma pendência, só me falar por favor.

**Comment by DanielDefault1** [comment]
@jonspada tudo bem, é que eu estou sentindo falta do título do email, que vocês falaram sobre no último checkpoint. Eu já esperava ver ele aqui. Algumas outras coisas para ajuste:
- Faça referência a estória da desativação da conta aqui, para que o time saiba de onde "surgirá" esse email na jornada do usuário. Que ele vai receber o email assim que encerrar o fluxo de desativação da conta.
- Vamos adicionar mais um critério de aceite relacionado com a usabilidade e acessibilidade do email: ele tem que ser responsivo e se adaptar ao tamanho da tela do usuário.

**Comment by anacvicente** [comment]
@PatrickOtero @luisa-spl 
esse fluxo mudou, vamos precisar ajustar o email, se não estiver em desenvolvimento, vamos incluir na próxima sprint
Poderia atualizar o status dela por favor?

**Comment by PatrickOtero** [comment]
Já está pronto, o Adriano já havia concluído esta tarefa. Quais seriam as mudanças, no caso? @anacvicente

**Comment by anacvicente** [comment]
@PatrickOtero, certo, Patrick. O fluxo de desativação/exclusão passou por uma atualização nas regras de negócio. 
Com a criação das funcionalidades de conta do mentorado e conta única, esses dois fluxos foram impactados, como já está pronto, vamos alterar nas próximas sprints então.

Basicamente o que muda é que a conta não é desativada, o mentor pode PAUSAR o perfil dele, mas manter a conta ativa usando o perfil de mentorado, se ele tiver os dois perfis. Se ele não tiver os dois perfis, continua acessando a conta, mas o perfil de mentor continua pausado até que ele remova a pausa

Acredito que o impacto maior é no front, pro back acredito que muda o nome e levemente a regra, mas posso validar isso com você para termos certeza. 

Esse ajuste precisa acontecer antes do lançamento. Vou chamar o Adriano pra falarmos sobre isso, ok?
